### PR TITLE
Swift app

### PR DESCRIPTION
**Pull request by pcuenca**
`generate.cpp` is a copy of `main.cpp` with some modifications necessary for a better experience:

- Model is loaded once.
- Generation callback to show progress.

Ideally we should merge those files and keep just one copy that supports both the command-line and the app use cases. We can do that in a followup PR.

Generation is using default parameters.

### CODE REVIEW

File: StarCoderApp/StarCoder/Assets.xcassets/AppIcon.appiconset/bloom-icon.png
Comment (NouamaneTazi) [inline StarCoderApp/StarCoder/Assets.xcassets/AppIcon.appiconset/bloom-icon.png:1]: Is this icon still used?

**Review by NouamaneTazi — CHANGES_REQUESTED** [review]
Very clean! One minor comment before we can merge this awesomeness

File: StarCoderApp/StarCoder/Assets.xcassets/AppIcon.appiconset/bloom-icon.png
Comment (pcuenca) [reply StarCoderApp/StarCoder/Assets.xcassets/AppIcon.appiconset/bloom-icon.png:1]: It's a placeholder until I can get a StarCoder version :) Slack comment [internal]: https://huggingface.slack.com/archives/C031GD0V5S5/p1684176403377149?thread_ts=1683292595.926539&cid=C031GD0V5S5

Comment (pcuenca) [reply StarCoderApp/StarCoder/Assets.xcassets/AppIcon.appiconset/bloom-icon.png:1]: I found one.

**Review by NouamaneTazi — APPROVED** [review]

**Pull request merged**